client.go

```diff
@@ -489,17 +491,40 @@ func (c *Client) httpRequest(method string, path string, data []byte, headers ma
 	var err error
 	for i := 0; i < maxHosts; i++ {
 		reader := bytes.NewReader(data)
-		// get a host from the cluster
-		host := c.cluster.Host()
-		if host == nil {
-			return nil, nil, ErrEmptyCluster
+		var host *URI
+
+		if useCoordinator {
+			c.coordinatorLock.RLock()
+			coordinatorURI := c.coordinatorURI
+			c.coordinatorLock.RUnlock()
+			if coordinatorURI == nil {
+				node, err := c.fetchCoordinatorNode()
+				if err != nil {
+					return nil, nil, errors.Wrap(err, "fetching coordinator node")
+				}
+				coordinatorURI = URIFromAddress(fmt.Sprintf("%s://%s:%d", node.Scheme, node.Host, node.Port))
+				c.coordinatorLock.Lock()
+				c.coordinatorURI = coordinatorURI
+				c.coordinatorLock.Unlock()
+			}
+			host = coordinatorURI
+		} else {
+			// get a host from the cluster
+			host = c.cluster.Host()
+			if host == nil {
+				return nil, nil, ErrEmptyCluster
+			}
```

Comment:
I think the logic in this whole if/else block should be wrapped up in cluster.Host and Host should be modified to take a boolean `useCoordinator` argument.

Reply:
I've thought about moving that logic to the cluster (probably `cluster.Host`) before but that creates a chicken-egg problem: we can't get the coordinator node without creating the client first; and creating the client requires a cluster. That if/else code is messy though, I'll move it to a separate function.